# HG changeset patch
# User Kirill Belokurov <kirill.belokurov@gmail.com>
# Date 1206631740 -7200
# Node ID 39383740e5296a82181f4857400d3f22f49f94fd
# Parent  a0b60931be1cb2eb9b4a4d412f1b6e32ca1c7349
take into account CPU endianness when converting 24bit=>16bit samples

diff -r a0b60931be1c -r 39383740e529 src/libxineadec/xine_lpcm_decoder.c
--- a/src/libxineadec/xine_lpcm_decoder.c	Thu Mar 27 17:27:12 2008 +0200
+++ b/src/libxineadec/xine_lpcm_decoder.c	Thu Mar 27 17:29:00 2008 +0200
@@ -189,11 +189,23 @@ static void lpcm_decode_data (audio_deco
 
     while (n >= 0) {
       if ( stream_be ) {
-	*d++ = s[0];
-	*d++ = s[1];
+	  if ( stream_be == this->cpu_be ) {
+	      *d++ = s[0];
+	      *d++ = s[1];
+	  } else {
+	      *d++ = s[1];
+	      *d++ = s[0];
+	  }
       } else {
-	*d++ = s[1];
-	*d++ = s[2];
+	  if ( stream_be == this->cpu_be ) {
+	      *d++ = s[1];
+	      *d++ = s[2];
+	  }
+	  else
+	  {
+	      *d++ = s[2];
+	      *d++ = s[1];
+	  }
       }
 
       s += 3;
